src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.c

```diff
@@ -457,54 +495,41 @@ void ObjKendoKanban_Update(Actor* thisx, PlayState* play) {
     ObjKendoKanban* this = THIS;
 
     this->actionFunc(this, play);
-
-    func_80B66304(this, play);
+    ObjKendoKanban_UpdateCollision(this, play);
 }
 
-#ifdef NON_MATCHING
 void ObjKendoKanban_Draw(Actor* thisx, PlayState* play) {
     ObjKendoKanban* this = THIS;
+
     s32 i;
-    Gfx* poly;
 
     OPEN_DISPS(play->state.gfxCtx);
 
     func_8012C28C(play->state.gfxCtx);
 
-    if (this->unk_30C == OBJKENDOKANBAN_F_0) {
+    if (this->boardFragments == PART_FULL) {
         gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
         gSPDisplayList(POLY_OPA_DISP++, gKendoKanbanDL);
     } else {
-        Matrix_RotateAxisS(this->unk_302, &this->unk_2F0, MTXMODE_APPLY);
-        Matrix_Translate(-this->unk_2E4.x, -this->unk_2E4.y, -this->unk_2E4.z, MTXMODE_APPLY);
-
+        // Fragments of boards are movable, so apply their rotation/translation, and display only the DLs which
+        // apply.
```

Comment:
This comment isn't really required, this is basically always what these matrix calls do in Draw functions.